networkx/algorithms/distance_measures.py

```diff
@@ -677,54 +680,64 @@ def resistance_distance(G, nodeA, nodeB, weight=None, invert_weight=True):
 
     Returns
     -------
-    rd : float
-       Value of effective resistance distance
+    rd : float (if `nodeA` and `nodeB` are given)
+       Resistance distance between `nodeA` and `nodeB`.
+         dictionary (if `nodeA` or `nodeB` is unspecified)
+       Dictionary of nodes with effective resistances as the value.
+
+    Raises
+    -------
+    NetworkXError
+        If `G` is disconnected.
+
+    NetworkXError
+        If `G` is a directed graph.
+
+    NetworkXError:
+        If `nodeA` is not in `G` or `nodeB` is not in `G`.
 
     Examples
     --------
     >>> G = nx.Graph([(1, 2), (1, 3), (1, 4), (3, 4), (3, 5), (4, 5)])
-    >>> nx.resistance_distance(G, 1, 3)
+    >>> round(nx.resistance_distance(G, 1, 3), 10)
     0.625
 
     Notes
     -----
-    Overviews are provided in [1]_ and [2]_. Additional details on computational
-    methods, proofs of properties, and corresponding MATLAB codes are provided
-    in [3]_.
+    The implementation is based on Theorem A in [2]_. Self-loops are ignored.
+    Multi-edges are contracted in one edge with weight equal to the harmonic sum of the weights.
 
     References
     ----------
     .. [1] Wikipedia
        "Resistance distance."
        https://en.wikipedia.org/wiki/Resistance_distance
-    .. [2] E. W. Weisstein
-       "Resistance Distance."
-       MathWorld--A Wolfram Web Resource
-       https://mathworld.wolfram.com/ResistanceDistance.html
-    .. [3] V. S. S. Vos,
-       "Methods for determining the effective resistance."
-       Mestrado, Mathematisch Instituut Universiteit Leiden, 2016
-       https://www.universiteitleiden.nl/binaries/content/assets/science/mi/scripties/master/vos_vaya_master.pdf
+    .. [2] D. J. Klein and M. Randic.
+        Resistance distance.
+        J. of Math. Chem. 12:81-95, 1993.
     """
     import numpy as np
-    import scipy as sp
 
+    if len(G) == 0:
+        msg = "Graph G must contain at least one node."
+        raise nx.NetworkXError(msg)
     if not nx.is_connected(G):
         msg = "Graph G must be strongly connected."
         raise nx.NetworkXError(msg)
-    elif nodeA not in G:
+    elif nx.is_directed(G):
+        msg = "Graph G must be undirected."
+        raise nx.NetworkXError(msg)
```

Comment:
This is what the `not_implemented_for` decorator does - I'd vote to leave that as-is.